cmd/aries-agent-mobile/pkg/api/storage.go

```diff
@@ -6,64 +6,95 @@ SPDX-License-Identifier: Apache-2.0
 
 package api
 
-// StoreIterator is the iterator for the latest snapshot of the underlying store.
-type StoreIterator interface {
-	// Next moves the iterator to the next key/value pair.
-	// It returns false if the iterator is exhausted.
-	Next() bool
-
-	// Free releases associated resources. Release should always success
-	// and can be called multiple times without causing error.
-	Free()
-
-	// Error returns any accumulated error. Exhausting all the key/value pairs
-	// is not considered to be an error.
-	Error() error
-
-	// Key returns the key of the current key/value pair, or nil if done.
-	// The caller should not modify the contents of the returned slice, and
-	// its contents may change on the next call to any 'seeks method'.
-	Key() []byte
-
-	// Value returns the value of the current key/value pair, or nil if done.
-	// The caller should not modify the contents of the returned slice, and
-	// its contents may change on the next call to any 'seeks method'.
-	Value() []byte
+import spi "github.com/hyperledger/aries-framework-go/spi/storage"
+
+// StorageProvider represents a storage provider.
+type StorageProvider interface {
```

Comment:
Isn't this defined in spi, imported above?

Reply:
My understanding from talking to @soluchok was that this duplication of interfaces is necessary for the mobile bindings. Perhaps @soluchok can elaborate?